fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/NavigationBar.kt

```diff
@@ -36,13 +43,26 @@ import org.mozilla.fenix.theme.Theme
  * Top-level UI for displaying the navigation bar.
  *
  * @param actionItems A list of [ActionItem] used to populate the bar.
- * @param menuButton A [MenuButton] to be used for [ItemType.MENU]
+ * @param menuButton A [MenuButton] to be used for [ItemType.MENU].
+ * @param browserStore A store to observe the number of opened tabs.
+ * @param browsingModeManager A helper class that provides access to the current [BrowsingMode].
  */
 @Composable
 fun NavigationBar(
     actionItems: List<ActionItem>,
     menuButton: MenuButton? = null,
+    browserStore: BrowserStore,
+    browsingModeManager: BrowsingModeManager,
 ) {
+    val isPrivate = browsingModeManager.mode.isPrivate
+    val tabCount = browserStore.observeAsComposableState { browserState ->
+        if (isPrivate) {
+            browserState.privateTabs.size
+        } else {
+            browserState.normalTabs.size
+        }
+    }.value ?: 0
```

Comment:
nit: there's a newer observe state function that returns a non-null value

Reply:
which one are you referring to?

Reply:
https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/lib/state/src/main/java/mozilla/components/lib/state/ext/ComposeExtensions.kt#62